optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/DefaultSolver.java

```diff
@@ -107,6 +108,21 @@ public Score getBestScore() {
         return solverScope.getBestScore();
     }
 
+    @Override
+    public String explainBestScore() {
+        Solution_ bestSolution = getBestSolution();
+        if (bestSolution == null) {
+            return null;
+        }
+        if (solverScope.getSolutionDescriptor().getScore(bestSolution) == null) {
```

Comment:
The javadoc says 
> null if {@link #getBestScore()} returns null

Reply:
Yes, but this style is intentional to avoid a race condition.
So I need to add a comment to explain that.

This method is thread-safe. Calling getBestScore() might yield a different score than the getBestSolution()'s because it was changed meanwhile.